ExampleApp-HandlingErrors/ViewController.swift

```diff
@@ -29,9 +29,60 @@ class ViewController: UIViewController {
         diceImage.layer.shadowRadius = 2
         diceImage.layer.shadowOffset = .zero
     }
+    
+    private var isRolling = false {
+        didSet {
+            UIView.animate(withDuration: 0.33) { [unowned self] in
+                diceImage.alpha = isRolling ? 0.5 : 1.0
+                diceImage.transform = isRolling ? CGAffineTransform(scaleX: 0.5, y: 0.5) :
+                    CGAffineTransform.identity
+            }
+            rollDiceButton.isEnabled = !isRolling
+        }
+    }
 
     @IBAction func rollDiceTapped(_ sender: Any) {
-        // Roll Dice
+        viewModel.rollDice()
+            .handleEvents(receiveSubscription: { [unowned self] _ in
+                isRolling = true
+            }, receiveCompletion: { [unowned self] _ in
+                isRolling = false
+            }, receiveCancel: { [unowned self] in
+                isRolling = false
+            })
```

Comment:
Again here, managing the state of `isRolling` would ideally be just a simple publisher that the view controller reacts to. That way the view model is in charge of the logic and can be tested outside of any UI.

Reply:
Agreed. Thanks.